txn_writer_test.go

```diff
@@ -567,3 +570,35 @@ func TestSafeWriter_VendorDotGitPreservedWithForceVendor(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+func TestCalculatePrune(t *testing.T) {
+	h := test.NewHelper(t)
+	defer h.Cleanup()
+
+	vendorDir := "vendor"
+	h.TempDir(vendorDir)
+	h.TempDir(filepath.Join(vendorDir, "github.com/keep/pkg/sub"))
+	h.TempDir(filepath.Join(vendorDir, "github.com/prune/pkg/sub"))
+
+	toKeep := []string{
+		filepath.FromSlash("github.com/keep/pkg"),
+		filepath.FromSlash("github.com/keep/pkg/sub"),
+	}
+
+	got, err := calculatePrune(h.Path(vendorDir), toKeep, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	sort.Sort(byLen(got))
+
+	want := []string{
+		h.Path(filepath.Join(vendorDir, "github.com/prune/pkg/sub")),
+		h.Path(filepath.Join(vendorDir, "github.com/prune/pkg")),
+		h.Path(filepath.Join(vendorDir, "github.com/prune")),
+	}
+
+	if !reflect.DeepEqual(want, got) {
+		t.Fatalf("expected %s, got %s", want, got)
```

Comment:
Let's make this more informative. Previously, I used to do the same but Sam asked me to make the failures more informative. Something like:
```
t.Fatalf("Calculated prune paths are not as expected: \n\t(GOT) %v\n\t(WNT) %v", got, want)
```